### ISSUE TITLE
DocumentServer doesn't run on CentOS7

### ISSUE THREAD

**Issue by smirta** [ONLYOFFICE/DocumentServer#832]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Cannot get onlyoffice-documentserver to install and run correctly due to binaries that are not executable by the system. Disabling selinux did not change the behaviour.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
Minimal CentOS7 installation and then
```
# yum install -y onlyoffice-documentserver
[...]
  Installing : onlyoffice-documentserver-5.5.0-165.x86_64                                                                                                                                                                             103/103 
Generating AllFonts.js, please wait.../bin/documentserver-generate-allfonts.sh: line 16: /var/www/onlyoffice/documentserver/server/tools/allfontsgen: Permission denied
Done
Generating presentation themes, please wait.../bin/documentserver-generate-allfonts.sh: line 32: /var/www/onlyoffice/documentserver/server/tools/allthemesgen: Permission denied
Done
[...]

# ls -al /var/www/onlyoffice/documentserver/server/tools/allfontsgen 
-rwxr-xr-x. 1 ds ds 82042  2. Mär 23:41 /var/www/onlyoffice/documentserver/server/tools/allfontsgen

# getfacl /var/www/onlyoffice/documentserver/server/tools/allfontsgen
getfacl: Removing leading '/' from absolute path names
# file: var/www/onlyoffice/documentserver/server/tools/allfontsgen
# owner: ds
# group: ds
user::rwx
group::r-x
other::r-x

# /var/www/onlyoffice/documentserver/server/tools/allfontsgen
-bash: /var/www/onlyoffice/documentserver/server/tools/allfontsgen: Permission denied

# bash /var/www/onlyoffice/documentserver/server/tools/allfontsgen
/var/www/onlyoffice/documentserver/server/tools/allfontsgen: /var/www/onlyoffice/documentserver/server/tools/allfontsgen: cannot execute binary file

# sudo -u ds /var/www/onlyoffice/documentserver/server/tools/allfontsgen
sudo: unable to execute /var/www/onlyoffice/documentserver/server/tools/allfontsgen: Permission denied

# sudo -u ds bash /var/www/onlyoffice/documentserver/server/tools/allfontsgen
/var/www/onlyoffice/documentserver/server/tools/allfontsgen: /var/www/onlyoffice/documentserver/server/tools/allfontsgen: cannot execute binary file

# file /var/www/onlyoffice/documentserver/server/tools/allfontsgen
/var/www/onlyoffice/documentserver/server/tools/allfontsgen: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux 2.6.24, BuildID[sha1]=86f0e5386c37199a22ce4f23bbe2205e1e691147, not stripped

# strace /var/www/onlyoffice/documentserver/server/tools/allfontsgen
strace -v /var/www/onlyoffice/documentserver/server/tools/allfontsgen
execve("/var/www/onlyoffice/documentserver/server/tools/allfontsgen", ["/var/www/onlyoffice/documentserv"...], ["LC_PAPER=de_CH.UTF-8", "XDG_SESSION_ID=1", "HOSTNAME=vm-0028.service.int.rab"..., "LC_MONETARY=de_CH.UTF-8", "SELINUX_ROLE_REQUESTED=", "TERM=xterm-256color", "SHELL=/bin/bash", "HISTSIZE=1000", "SSH_CLIENT=10.136.0.2 60570 22", "SELINUX_USE_CURRENT_RANGE=", "LC_NUMERIC=de_CH.UTF-8", "SSH_TTY=/dev/pts/0", "USER=root", "LS_COLORS=rs=0:di=38;5;27:ln=38;"..., "COBBLER_SERVER=vm-0001.service.i"..., "MAIL=/var/spool/mail/root", "PATH=/usr/local/sbin:/usr/local/"..., "PWD=/root", "XMODIFIERS=@im=ibus", "LANG=en_US.UTF-8", "LC_MEASUREMENT=de_CH.UTF-8", "SELINUX_LEVEL_REQUESTED=", "HISTCONTROL=ignoredups", "SHLVL=1", "HOME=/root", "LOGNAME=root", "SSH_CONNECTION=10.136.0.2 60570 "..., "LESSOPEN=||/usr/bin/lesspipe.sh "..., "XDG_RUNTIME_DIR=/run/user/0", "LC_TIME=de_CH.UTF-8", "_=/usr/bin/strace"]) = -1 EACCES (Permission denied)
write(2, "strace: exec: Permission denied\n", 32strace: exec: Permission denied
) = 32
exit_group(1)                           = ?
+++ exited with 1 +++
```
It affects all binaries. Due to not being able to run the binaries, all services started by supervisord fail. Seems like some dependencies are missing or 5.5.0 is not supposed to run on CentOS7 at all. Even when https://helpcenter.onlyoffice.com/de/server/linux/document/linux-installation-centos.aspx says otherwise.

```
# objdump -s --section .comment /var/www/onlyoffice/documentserver/server/DocService/docservice 

/var/www/onlyoffice/documentserver/server/DocService/docservice:     file format elf64-x86-64

Contents of section .comment:
 0000 4743433a 2028474e 55292034 2e342e37  GCC: (GNU) 4.4.7
 0010 20323031 32303331 33202852 65642048   20120313 (Red H
 0020 61742034 2e342e37 2d323329 00474343  at 4.4.7-23).GCC
 0030 3a202847 4e552920 382e332e 31203230  : (GNU) 8.3.1 20
 0040 31393033 31312028 52656420 48617420  190311 (Red Hat 
 0050 382e332e 312d3329 00                 8.3.1-3).       

```

**What is the expected behavior?**
Binaries are executable.

**Did this work in previous versions of DocumentServer?**
n/a

**DocumentServer version:**
5.5.0

**Operating System:**
CentOS Linux release 7.6.1810 (Core)

**Comment by ShockwaveNN** [comment]
Hi, i just checked installation on CentOS-7-x86_64-DVD-1804.iso with latest updates:
```
# cat /etc/centos-release
CentOS Linux release 7.7.1908 (Core)
```

And everything seems fine
Could you show output of 
```
uname -a
```

**Comment by smirta** [comment]
Thanks for looking into this. 
```
Linux localhost.localdomain 3.10.0-957.21.3.el7.x86_64 #1 SMP Tue Jun 18 16:35:19 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

Is the kernel too old? The required kernel for compiling (https://helpcenter.onlyoffice.com/de/server/linux/document/compile-source-code.aspx) is 3.13 or later. But the latest kernel for CentOS7 is the kernel you have and also not 3.13 or later

**Comment by ShockwaveNN** [comment]
@smirta No, everything seems fine. Just check that you're not on ARM or x32 )

Could you provide image of CentOS you're using. Is it have some GUI?

**Comment by smirta** [comment]
No, it's a CentOS7 Minimal installation from our repositories that we sync with the official mirrors every day and freeze when there is a maintenance window. The last freeze was unfortunately June 23rd 2019. Maybe it's because it's a minimal installation?

**Comment by ShockwaveNN** [comment]
@smirta I'm not very familiar with centos. 
Could I install this minimal installation from official centos dvd image?

**Comment by smirta** [comment]
Sure, on any mirror you should have alternative isos. Go to http://isoredirect.centos.org/centos/7/isos/x86_64/ and choose a mirror nearby. There you should find an image "CentOS-7-x86_64-Minimal-1908.iso". 

Our iso you can't find on every mirror anymore. You could get it here: http://mirrors.oit.uci.edu/centos/7.6.1810/isos/x86_64/

**Comment by ShockwaveNN** [comment]
@smirta checked centos installed by downlading this torrent http://mirrors.oit.uci.edu/centos/7.6.1810/isos/x86_64/CentOS-7-x86_64-Minimal-1810.torrent

And then installing DocumentServer using this script - https://github.com/onlyoffice-testing-robot/documentserver-install-from-manual/ (should be same commands as in instruction) and 
```
[virtualbox@localhost Centos7]$ sudo supervisorctl status all
ds:converter                     RUNNING   pid 15196, uptime 0:00:27
ds:docservice                    RUNNING   pid 15195, uptime 0:00:27
ds:example                       RUNNING   pid 15200, uptime 0:00:27
ds:metrics                       RUNNING   pid 15193, uptime 0:00:27
ds:spellchecker                  RUNNING   pid 15194, uptime 0:00:27
```
Seems that everything is fine.
Maybe you can provide some VM image with already installed DocumentServer?

**Comment by smirta** [comment]
what the heck... let me check. On what services does DocService depend? Rabbitmq? Anything else? We have rabbitmq and postgresql clusters. Probably I've messed up. I'll check later and let you know. Sorry for the trouble.

**Comment by ShockwaveNN** [comment]
@smitra take a look at logs at `/var/log/onlyoffice` for some errors. Don't know why didn't asked this first :(

**Comment by smirta** [comment]
Well, I was looking there but wasn't able to find anything suspicious apart from the described errors while installing and in supervisor-log.

**Comment by smirta** [comment]
does not work on my machine. This time I used your script. Can you provide the list of installed packages? I suspect something's wrong there.

And there is a difference between script and documentation. In the script onlyoffice-documentserver-ie gets installed. In the documentation onlyoffice-documentserver.

**Comment by ShockwaveNN** [comment]
> In the script onlyoffice-documentserver-ie gets installed. In the documentation onlyoffice-documentserver.

Yeah, but technically this is a same product, you can change script to install another variant.

```
Requires: nginx >= 1.3.13, postgresql >= 9.1, wget, supervisor >= 3.0b2, libstdc++ >= 4.8.4, libcurl, libxml2, zlib, libXScrnSaver, atk, gtk3, xorg-x11-server-Xvfb, libXtst, GConf2, alsa-lib, pwgen, libselinux-utils, liberation-mono-fonts, logrotate, ca-certificates
```
This is a dependencies of package

**Comment by smirta** [comment]
woki. Unfortunately there are no logs since the scripts are not executable. I have also disabled selinux. I really have no idea what's wrong here. I know the dependencies but I suspect there is something else I didn't think of. That's why I was asking for a list of the installed packages. I'll install the image and have a look at the differences. Thanks a lot for your support. Very appreciated

**Comment by ShockwaveNN** [comment]
@smirta This is result of executing `yum list installed`
https://gist.github.com/ShockwaveNN/8cf6870dafdd031b34ed652492e3c0c8

I can send you VirtualBox image with documentserver installed if you want

**Comment by smirta** [comment]
Oh, how I love to waste lifetime. After spending another evening debugging, after re-installing a million times, adjusting all the packages to the exact same version as yours, analyzing the differences in /etc, I arrived at fstab. And what do I see there? You probably know what comes next. "noexec" ... Our systems are configured to mount /var with noexec flag set. Somebody invented this some time ago and I forgot about it. After I realized, my mind was close to a kernel panic.

Thank you very much for your outstanding support and sorry for wasting your lifetime, too. Have a good one
